lib/chef/resource/yum_package.rb

```diff
@@ -28,6 +28,7 @@ def initialize(name, run_context=nil)
         super
         @flush_cache = { :before => false, :after => false }
         @allow_downgrade = false
+        @yum_binary = ::File.exist?("/usr/bin/yum-deprecated") ? "yum-deprecated" : "yum"
```

Comment:
Well it felt like it was a default value, but now that I think about it, that moves it to compile time rather than converge time so any resources that would install yum-deprecated would not have run yet, so I should probably move that...